### PR TITLE
Feature/homepage product filter

### PR DESCRIPTION
**Pull request by govindappaarun**
- Added Home Page
- Category browsing 
- Refactor filter & add a hook for filtering

### CODE REVIEW

**Review by BISINIGISHIVANI — APPROVED** [review]
Good commit messeges and folder structure.
and small suggestion In home page you can use signup instead create account which is also familar user

File: src/pages/home/home.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+              learn with expert anytime, anywhere.
+            </Typography>
+            <Button
+              onClick={() => navigate("/signup", { replace: true })}
+              className="btn btn-warning"
+            >
+              Create Account
+            </Button>
+          </Box>
+          {/* <Box grow={1} className="hero-section-right">
```
Comment (BISINIGISHIVANI) [inline src/pages/home/home.tsx]: you can remove commented  code if not using that code

Comment (BISINIGISHIVANI) [reply src/pages/home/home.tsx]: clicking on category pages getting 404 no matching route found. you can fix it.

File: src/pages/home/home.styled.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    display: flex;
+  }
+
+  .hero-section-left {
+    align-items: center;
+    justify-content: center;
+  }
+
+  .hero-section-left .h1 {
+    /* max-width: 16ch; */
```
Comment (MayankKumar10) [inline src/pages/home/home.styled.ts:24]: remove these comment

File: src/pages/home/home.tsx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            <img
+              loading="lazy"
+              className="img-responsive"
+              src={Hero}
+              alt="hero sectionimage"
```
Comment (MayankKumar10) [inline src/pages/home/home.tsx]: remove the commented code

File: src/pages/product-list/Components/ProductFilter.jsx [hunk 4]
```diff
@@ -1,14 +1,39 @@
...
+      dispatch({
+        type: "SET_CATEGORY",
+        payload: { item: { name: params.categoryName, checked: true } },
+      });
+    }
+  }, [params, categories]);
+
+  useEffect(() => {
+    productService.getCategories().then((response) => {
+      console.log({ response });
```
Comment (MayankKumar10) [inline src/pages/product-list/Components/ProductFilter.jsx:29]: remove console log

**Review by MayankKumar10 — APPROVED** [review]
Look into these everything else Looks good to me you can merge it.

**Pull request merged**